manta-util/src/iter/mod.rs

```diff
@@ -40,6 +49,16 @@ pub use select_all::SelectAll;
 
 /// Iterator Extensions
 pub trait IteratorExt: Iterator {
+    /// Searches for an element of an iterator that the `finder` matches with, returning the mapped
+    /// value from `f`.
+    #[inline]
+    fn find_with<T, F, R>(&mut self, finder: &mut Finder<T>, f: F) -> Option<R>
```

Comment:
Maybe it's more intuitive if we just take a state as parameter instead of `finder`?

Reply:
Taking a `Finder` is more general since we don't know if the state has been found yet or not.